packages/engine-render/src/components/docs/layout/block/paragraph/language-ruler.ts

```diff
@@ -35,19 +35,27 @@ export function otherHandler(
 ) {
     const glyphGroup = [];
     let step = 0;
+    let src = charArray;
 
-    for (let i = 0; i < charArray.length; i++) {
-        const newChar = charArray[i];
+    while (src.length) {
+        const char = src.match(/^[\s\S]/gu)?.[0];
 
-        if (hasSpace(newChar) || startWithEmoji(charArray.substring(i))) {
+        if (char == null) {
```

Comment:
Remove `==` from our codebase.

Reply:
Shouldn `== null` be allowed? because the phonetics here are very clear, which is `=== undefined` or `=== null`.

Reply:
![image](https://github.com/user-attachments/assets/b57a88af-a4c6-48fc-8274-937c0f2e6214)

Like this. The core idea is to avoid implicit type casting.

Reply:
其实我想表达的是，使用 == null,应该就是明确 `=== null || === undefined` 的意思，在其他地方是不建议使用 ==。之前还看了一篇文章，你所不知道的 JavaScript 作者写的，专门还说了这个问题，我找找